code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -58,106 +58,91 @@
-
-	for(var/word in words)
-		play_vox_word(word, src.z, null)
-
-
-/proc/play_vox_word(word, z_level, mob/only_listener)
-
-	word = lowertext(word)
+	var/emotion_to_use = "Neutral"
+	switch(character_to_use)
+		if("GLaDOS (Homicidal)")
+			emotion_to_use = "Homicidal"
+			character_to_use = "GLaDOS"
+		if("GLaDOS (Neutral)")
+			character_to_use = "GLaDOS"
+		if("Twilight Sparkle (Happy)")
+			emotion_to_use = "Happy"
+			character_to_use = "Twilight Sparkle"
+		if("Twilight Sparkle (Neutral)")
+			character_to_use = "Twilight Sparkle"
+	last_announcement = message
 
-	if(GLOB.vox_sounds[word])
+	announcing_vox = world.time + VOX_DELAY
 
-		var/sound_file = GLOB.vox_sounds[word]
-		var/sound/voice = sound(sound_file, wait = 1, channel = CHANNEL_VOX)
+	log_game("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	message_admins("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	play_vox_word(message, character_to_use, emotion_to_use, src, src.z, null)
+
+
+/proc/play_vox_word(message, character, emotion, mob/living/speaker, z_level, mob/only_listener)
+	var/api_url = "https://api.fifteen.ai/app/getAudioFile"
+
+	var/datum/http_request/req = new()
+	var/vox_audio_number = rand(1,9999)
+	req.prepare(RUSTG_HTTP_METHOD_POST, api_url, "{\"character\":\"[character]\",\"text\":\"[message]\",\"emotion\":\"[emotion]\"}", list("Content-Type" = "application/json", "User-Agent" = "/tg/station 13 server"), json_encode(list("output_filename" = "data/vox_[vox_audio_number].wav")))
+	req.begin_async()
+	UNTIL(req.is_complete())
+	var/datum/http_response/res = req.into_response()
+	if(res.status_code == 200)
+		log_game("[key_name(speaker)] finished making a 15.AI announcement with the following message: [message]")
+		message_admins("[key_name(speaker)] finished making a 15.AI announcement with the following message: [message]")
+		speaker.say(";[message]")
+		var/sound/voice = sound("data/vox_[vox_audio_number].wav", wait = 1, channel = CHANNEL_VOX)
 		voice.status = SOUND_STREAM
-
  		// If there is no single listener, broadcast to everyone in the same z level
 		if(!only_listener)
 			// Play voice for all mobs in the z level
 			for(var/mob/M in GLOB.player_list)
-				if(M.can_hear() && (M.client.prefs.toggles & SOUND_ANNOUNCEMENTS))
-					var/turf/T = get_turf(M)
-					if(T.z == z_level)
-						SEND_SOUND(M, voice)
+				var/turf/T = get_turf(M)
+				if(T.z == z_level)
+					SEND_SOUND(M, voice)
 		else
 			SEND_SOUND(only_listener, voice)
+		fdel("data/vox_[vox_audio_number].wav")
 		return 1
+	else
+		log_game("[key_name(speaker)] failed to produce a 15.AI announcement due to an error. Error code: [res.status_code]")
+		message_admins("[key_name(speaker)] failed to produce a 15.AI announcement due to an error. Error code: [res.status_code]")
+		to_chat(speaker, "The speech synthesizer failed to return audio. Please try again.")

```

Comment:
The request must not return a status code, the response could fail entirely in which case additional logging would be neat.